pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -298,12 +314,22 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	}
 
 	if osdProps.onPVC() && osd.CVMode == "raw" {
-		initContainers = append(initContainers, c.getPVCInitContainerActivate(osdDataDirPath, osdProps))
-		if osdProps.onPVCWithMetadata() {
-			initContainers = append(initContainers, c.getPVCMetadataInitContainerActivate(osdDataDirPath, osdProps))
+		if isEncryptedPVC {
+			// Add a new init container to open the encrypted disk!
+			initContainers = append(initContainers, c.getPVCEncryptionOpenInitContainerActivate(osdProps))
+			initContainers = append(initContainers, c.getPVCEncryptionInitContainerActivate(osdDataDirPath, osdProps))
+			// DON4T FORGET METADATA DEV
```

Comment:
TODO?

Reply:
Yes, just need testing and more init activate container :)

Reply:
As discussed, let's follow up!